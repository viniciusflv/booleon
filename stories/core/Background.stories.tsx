import { Background } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Background',
  component: Background,
};

export const Default: FC = () => (
  <Background
    gy_grey
    bg_black
    bg_img="https://www.w3schools.com/cssref/img_tree.gif"
    style={{ width: '500px', height: '1000px' }}>
    <span style={{ color: 'white' }}>texto</span>
  </Background>
);
