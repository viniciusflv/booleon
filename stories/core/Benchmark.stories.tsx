// import React, { useState, useEffect } from 'react';

// import { withKnobs } from '@storybook/addon-knobs';

// import { View } from '../../packages/base/src';
// import benchmarkProps from '../benchmarkProps';

// export default {
//   title: 'Core / Benchmark',
//   component: View,
//   decorators: [withKnobs],
// };

// const Div = (props) => <div {...props} />;

// const DivStyled = View.styled(Div);

// const Styled = ({ bool }) => (
//   <>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     {/* <DivStyled {...benchmarkProps(bool)}>styled</DivStyled>
//     <DivStyled {...benchmarkProps(bool)}>styled</DivStyled> */}
//   </>
// );

// const Main = ({ bool }) => (
//   <>
//     <View.div content={true} {...benchmarkProps(bool)}>
//       div
//     </View.div>
//     <View.div {...benchmarkProps(bool)}>div</View.div>
//     <View.div {...benchmarkProps(bool)}>div</View.div>
//     {/* <View.div {...benchmarkProps(bool)}>div</View.div>
//     <View.div {...benchmarkProps(bool)}>div</View.div> */}
//   </>
// );

// const Multiple = ({ bool }) => (
//   <View.multiple {...benchmarkProps(bool)}>
//     <div>multiple</div>
//     <div>multiple</div>
//     <div>multiple</div>
//     {/* <div>multiple</div>
//     <div>multiple</div> */}
//   </View.multiple>
// );

// export const Default = () => {
//   const [bool, setBool] = useState(true);
//   useEffect(() => {
//     const timeout = setTimeout(() => setBool(!bool), 1000);
//     return () => clearTimeout(timeout);
//   }, [bool]);

//   return (
//     <>
//       <Multiple bool={bool} />
//       <Main bool={bool} />
//       <Styled bool={bool} />
//     </>
//   );
// };
