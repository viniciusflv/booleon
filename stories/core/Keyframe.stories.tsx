// import React, { FC } from 'react';

// import { withKnobs } from '@storybook/addon-knobs';

// import { View } from '../../packages/base/src';
// import { Keyframe } from '../../packages/base/src/Keyframe';

// export default {
//   title: 'Core / Keyframe',
//   component: View,
//   decorators: [withKnobs],
// };

// const Keyframes: FC<any> = ({ children }) => (
//   <>
//     <Keyframe.provider scale>
//       <Keyframe.state sl_50 />
//       <Keyframe.state sl_100 />
//     </Keyframe.provider>
//     <Keyframe.provider spin>
//       <Keyframe.state rt_360 />
//     </Keyframe.provider>
//     <Keyframe.provider turn>
//       <Keyframe.state rt_180 />
//     </Keyframe.provider>
//     <Keyframe.provider drop>
//       <Keyframe.state top />
//       <Keyframe.state top_full />
//     </Keyframe.provider>
//     <Keyframe.provider left>
//       <Keyframe.state left_neg />
//       <Keyframe.state left />
//     </Keyframe.provider>
//     <Keyframe.provider right>
//       <Keyframe.state right_neg />
//       <Keyframe.state right />
//     </Keyframe.provider>
//     {children}
//   </>
// );
// export const Default = () => {
//   return (
//     <Keyframes>
//       <View.div relative>
//         <View.multiple
//           h_50
//           w_50
//           ani_duration_3s
//           ani_iteration_infinite
//           ani_forwards
//           ani_ease_in_out>
//           <View.div bg_fd0f00 kf_scale />
//           <View.div bg_5f0f55 kf_spin />
//           <View.div bg_505f50 kf_turn />
//           <View.div bg_0d0f50 kf_left relative />
//           <View.div bg_0d5ff0 kf_right relative />
//           <View.div bg_0ffd5f kf_drop absolute />
//         </View.multiple>
//       </View.div>
//     </Keyframes>
//   );
// };
