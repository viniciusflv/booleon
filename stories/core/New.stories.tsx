import React from 'react';

import { styledBooleon, useTheme } from '../../packages/react/src';

import {
  spacing,
  sizing,
  background,
  border,
  shadow,
} from '../../packages/modules/src';

export default {
  title: 'Core / New',
};

const Div = styledBooleon.div(background, spacing, sizing, border, shadow);

export const Default = () => {
  const { toggleTheme } = useTheme();
  return (
    <Div
      sd_8
      h_400
      w_400
      b_edge
      btr_rounded
      bbl_rounded
      bg_d5d5d5
      dark__bg_2f2f2f
      onClick={toggleTheme}
    />
  );
};
