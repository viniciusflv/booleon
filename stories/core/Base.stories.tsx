import { Base } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Base',
  component: Base,
};

export const NoProps: FC<typeof Base> = () => <Base />;
