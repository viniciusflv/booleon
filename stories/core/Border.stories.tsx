import { Border } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Border',
  component: Border,
};

export const Default: FC = () => <Border />;
