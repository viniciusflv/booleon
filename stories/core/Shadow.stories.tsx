import { Container } from '../../packages/core/src';
import React, { FC } from 'react';

export default {
  title: 'Core | Container',
  component: Container,
};

export const Default: FC = () => (
  <div style={{ display: 'flex' }}>
    <div style={{ height: '1rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '2rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '4rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '8rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '16rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '32rem', margin: '10px', background: 'red', width: '10px' }} />
    <div style={{ height: '64rem', margin: '10px', background: 'red', width: '10px' }} />
  </div>
);
