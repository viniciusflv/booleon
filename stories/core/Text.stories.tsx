import { Text } from '../../packages/core/src';
import { text, withKnobs } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import React, { FC } from 'react';

export default {
  title: 'Core | Text',
  component: Text,
  decorators: [withA11y, withKnobs],
};

export const Default: FC = () => {
  const props = text('Props', '')
    .split(/\s/)
    .reduce((acc, prop) => ({ ...acc, [prop]: true }), {});
  return <Text {...props}>{text('Text', 'Hello World')}</Text>;
};
