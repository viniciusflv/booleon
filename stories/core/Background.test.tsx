import 'jest-styled-components';
import { Background } from '../../packages/core/src';
import React from 'react';
import renderer from 'react-test-renderer';

const injectPseudoElement = (pseudo: string, component: any) => {
  component.props.className = component.props.className
    .split(/\s/)
    .map((classes: string) => ` ${classes}::${pseudo}`)
    .join('');
  return component;
};

describe('Background', () => {
  test('bg_img', () => {
    const bg_img = 'https://www.w3schools.com/cssref/img_tree.gif';
    const component = renderer.create(<Background bg_img={bg_img} />).toJSON();
    const before = injectPseudoElement('before', component);
    expect(before).toHaveStyleRule('background-image', `url(${bg_img})`);
  });

  describe('Repeat', () => {
    test('bg_repeat', () => {
      const component = renderer.create(<Background bg_repeat />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-repeat', 'repeat');
    });

    test('bg_norepeat', () => {
      const component = renderer.create(<Background bg_norepeat />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-repeat', 'no-repeat');
    });
  });

  describe('Size', () => {
    test('bg_auto', () => {
      const component = renderer.create(<Background bg_auto />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-size', 'auto');
    });

    test('bg_cover', () => {
      const component = renderer.create(<Background bg_cover />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-size', 'cover');
    });

    test('bg_cover', () => {
      const component = renderer.create(<Background bg_cover />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-size', 'cover');
    });
  });

  describe('Attachment', () => {
    test('bg_fixed', () => {
      const component = renderer.create(<Background bg_fixed />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-attachment', 'fixed');
    });

    test('bg_local', () => {
      const component = renderer.create(<Background bg_local />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-attachment', 'local');
    });

    test('bg_scroll', () => {
      const component = renderer.create(<Background bg_scroll />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-attachment', 'scroll');
    });
  });

  describe('Color', () => {
    test('bg_black', () => {
      const component = renderer.create(<Background bg_black />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-color', '#000');
    });

    test('bg_white', () => {
      const component = renderer.create(<Background bg_white />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-color', '#fff');
    });

    test('bg_grey', () => {
      const component = renderer.create(<Background bg_grey />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('background-color', '#cbd5e0');
    });
  });

  describe('Opacity', () => {
    test('bg_op_0', () => {
      const component = renderer.create(<Background bg_op_0 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '0');
    });

    test('bg_op_10', () => {
      const component = renderer.create(<Background bg_op_10 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.1');
    });

    test('bg_op_20', () => {
      const component = renderer.create(<Background bg_op_20 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.2');
    });

    test('bg_op_30', () => {
      const component = renderer.create(<Background bg_op_30 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.3');
    });

    test('bg_op_40', () => {
      const component = renderer.create(<Background bg_op_40 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.4');
    });

    test('bg_op_50', () => {
      const component = renderer.create(<Background bg_op_50 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.5');
    });

    test('bg_op_60', () => {
      const component = renderer.create(<Background bg_op_60 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.6');
    });

    test('bg_op_70', () => {
      const component = renderer.create(<Background bg_op_70 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.7');
    });

    test('bg_op_80', () => {
      const component = renderer.create(<Background bg_op_80 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.8');
    });

    test('bg_op_90', () => {
      const component = renderer.create(<Background bg_op_90 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '.9');
    });

    test('bg_op_100', () => {
      const component = renderer.create(<Background bg_op_100 />).toJSON();
      const before = injectPseudoElement('before', component);
      expect(before).toHaveStyleRule('opacity', '1');
    });
  });
});

describe('Gradient', () => {
  describe('Opacity', () => {
    test('g_op_0', () => {
      const component = renderer.create(<Background g_op_0 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '0');
    });

    test('g_op_10', () => {
      const component = renderer.create(<Background g_op_10 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.1');
    });

    test('g_op_20', () => {
      const component = renderer.create(<Background g_op_20 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.2');
    });

    test('g_op_30', () => {
      const component = renderer.create(<Background g_op_30 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.3');
    });

    test('g_op_40', () => {
      const component = renderer.create(<Background g_op_40 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.4');
    });

    test('g_op_50', () => {
      const component = renderer.create(<Background g_op_50 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.5');
    });

    test('g_op_60', () => {
      const component = renderer.create(<Background g_op_60 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.6');
    });

    test('g_op_70', () => {
      const component = renderer.create(<Background g_op_70 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.7');
    });

    test('g_op_80', () => {
      const component = renderer.create(<Background g_op_80 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.8');
    });

    test('g_op_90', () => {
      const component = renderer.create(<Background g_op_90 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '.9');
    });

    test('g_op_100', () => {
      const component = renderer.create(<Background g_op_100 />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule('opacity', '1');
    });
  });

  describe('X', () => {
    test('gx_black', () => {
      const component = renderer.create(<Background gx_black />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to right,#000,transparent,#000)',
      );
    });

    test('gx_white', () => {
      const component = renderer.create(<Background gx_white />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to right,#fff,transparent,#fff)',
      );
    });

    test('gx_grey', () => {
      const component = renderer.create(<Background gx_grey />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to right,#cbd5e0,transparent,#cbd5e0)',
      );
    });
  });

  describe('Y', () => {
    test('gy_black', () => {
      const component = renderer.create(<Background gy_black />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to top,#000,transparent,#000)',
      );
    });

    test('gy_white', () => {
      const component = renderer.create(<Background gy_white />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to top,#fff,transparent,#fff)',
      );
    });

    test('gy_grey', () => {
      const component = renderer.create(<Background gy_grey />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to top,#cbd5e0,transparent,#cbd5e0)',
      );
    });
  });

  describe('Top', () => {
    test('gt_black', () => {
      const component = renderer.create(<Background gt_black />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to top,transparent,#000)',
      );
    });

    test('gt_white', () => {
      const component = renderer.create(<Background gt_white />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to top,transparent,#fff)',
      );
    });

    test('gt_grey', () => {
      const component = renderer.create(<Background gt_grey />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to top,transparent,#cbd5e0)',
      );
    });
  });

  describe('Bottom', () => {
    test('gb_black', () => {
      const component = renderer.create(<Background gb_black />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to bottom,transparent,#000)',
      );
    });

    test('gb_white', () => {
      const component = renderer.create(<Background gb_white />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to bottom,transparent,#fff)',
      );
    });

    test('gb_grey', () => {
      const component = renderer.create(<Background gb_grey />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to bottom,transparent,#cbd5e0)',
      );
    });
  });

  describe('Left', () => {
    test('gl_black', () => {
      const component = renderer.create(<Background gl_black />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to left,transparent,#000)',
      );
    });

    test('gl_white', () => {
      const component = renderer.create(<Background gl_white />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to left,transparent,#fff)',
      );
    });

    test('gl_grey', () => {
      const component = renderer.create(<Background gl_grey />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to left,transparent,#cbd5e0)',
      );
    });
  });

  describe('Right', () => {
    test('gr_black', () => {
      const component = renderer.create(<Background gr_black />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to right,transparent,#000)',
      );
    });

    test('gr_white', () => {
      const component = renderer.create(<Background gr_white />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to right,transparent,#fff)',
      );
    });

    test('gr_grey', () => {
      const component = renderer.create(<Background gr_grey />).toJSON();
      const after = injectPseudoElement('after', component);
      expect(after).toHaveStyleRule(
        'background-image',
        'linear-gradient(to right,transparent,#cbd5e0)',
      );
    });
  });
});
