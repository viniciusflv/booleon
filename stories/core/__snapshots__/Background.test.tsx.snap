// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background bg_img 1`] = `
.c0 {
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: -1;
}

.c0::before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: -1;
  background-image: url(https://www.w3schools.com/cssref/img_tree.gif);
}

<div
  className="c0"
/>
`;
